### PR TITLE
Adds Nodejs support with minimal changes

### PR DESCRIPTION
**Pull request by zol**
- Adds Chai as an Npm dependency as CDN no longer server the requested Chai version. Comments out CORS test that doesn't pass.
- Ports existing Chai tests to jest for Node.
- Makes minimal changes to resemble.js to support both Node and browsers.
- Adds Node Buffer support along with tests and a Promise based `compareImages` convenience function.

### CODE REVIEW

**Comment by jamescryer** [comment]
`npm run test` fails for me  with 

```
 FAIL  .\resemble.test.js
  ? resemble > base64

    Cannot find module 'canvas' from 'resemble.js'

      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:179:17)
      at loadImageData (resemble.js:164:23)
      at compare (resemble.js:583:4)
      at wrapper (resemble.js:676:7)
      at Object.onComplete (resemble.js:679:6)
```

**Comment by zol** [comment]
@jamescryer Under Node, the canvas Npm module is used instead of the native canvas support built into the browser. Unfortunately the package relies on some native system libraries however they're rather easy to install (I added a link to the readme for instructions [here](https://www.npmjs.com/package/canvas), if you're on OSX the command is just `brew install pkg-config cairo libpng jpeg giflib`). 

I've also made it so that the `canvas` module is a peer dependency so you'll need to manually `npm install canvas`. The reason for this is to ensure module dependency behavior is backwards compatible with the way things are now, i.e browser users don't have to install the canvas dependency which comes with the additional step of installing the system dependency packages.

So to sum it up, you'll most likely need to run 

```bash
brew install pkg-config cairo libpng jpeg giflib
npm install canvas
npm run test
```

**Comment by zol** [comment]
@jamescryer Is this PR something you're willing to take? I'd be happy to help get it to the state where you're ok to merge it.

**Comment by jamescryer** [comment]
I'll review again in a couple weeks (I only have so much time to spend on this).  I'll need to see clear instructions on usage for windows/linux/mac, ideally added to the README as part of the PR.

Thanks

**Comment by zol** [comment]
Gotcha. I've updated the README with detailed instructions. I've included a complete installation example for OSX but am still linking to the binary package prerequisite installation instructions on canvas's site in case they change in the future.

**Comment by wswebcreation** [comment]
@zol 

Nice to have the Node support, on the other hand, what would be the benefit of this, only for local development? Or maybe for a npm package?
It's not that I want to judge your work, more to see the benefit of this. 

Maybe we can also make it canvas independent so it's also OS independent?

**Comment by zol** [comment]
@wswebcreation There are many good reasons for doing image comparison on the server (hence support for Node). At [Chromatic](http://hichroma.com) we're building a tool for UI testing that takes screen captures of UI components (quite possibly hundreds at a time) and compares them to a previously know good state. It brings to the user's attention any comparisons that fail. A browser is used only to view the failed comparisons. It would be unfeasible from a user experience perspective to try to do these comparisons in the browser.

Canvas is already OS independent, i.e it has support for all major OSs. Because it's doing heavy image manipulation, it benefits from using fast native libraries rather than being a pure Javascript package.

**Comment by jamescryer** [comment]
I merged in the PR, and made some small changes.  Thanks!

**Comment by zol** [comment]
Fantastic, thanks @jamescryer!

**Comment by mattblang** [comment]
@zol It looks like on OSX you also need to `brew install zlib` then `export PKG_CONFIG_PATH=/usr/local/opt/zlib/lib/pkgconfig` in your `bash_profile`.

**Comment by zol** [comment]
Not sure @mattblang , I've never had to do that.

**Comment by mattblang** [comment]
@zol Here is the error trace of what happens if I don't do that:

```
npm i

> fsevents@1.1.2 install /path/to/project/spec/puppeteer/node_modules/fsevents
> node install

[fsevents] Success: "/path/to/project/spec/puppeteer/node_modules/fsevents/lib/binding/Release/node-v57-darwin-x64/fse.node" already installed
Pass --update-binary to reinstall or --build-from-source to recompile

> canvas@1.6.7 install /path/to/project/spec/puppeteer/node_modules/canvas
> node-gyp rebuild

Package zlib was not found in the pkg-config search path.
Perhaps you should add the directory containing `zlib.pc'
to the PKG_CONFIG_PATH environment variable
Package 'zlib', required by 'libpng', not found
gyp: Call to './util/has_lib.sh freetype' returned exit status 0 while in binding.gyp. while trying to load binding.gyp
```
